File: BuildJavaUI/src/org/kbinani/windows/forms/BListViewItem.java
package org.kbinani.windows.forms;

import java.util.Vector;
import javax.swing.table.TableModel;

public class BListViewItem implements Cloneable{
    private Vector<String> subItems = new Vector<String>();
    private Object tag;
    protected TableModel tmodel = null;
    protected int row;
    //private String group = "";
    private String name = "";
    
    public BListViewItem( String[] values ){
        subItems.clear();
        for( int i = 0; i < values.length; i++ ){
            subItems.add( values[i] );
        }
    }

    protected BListViewItem( TableModel tmodel, int row ){
        this.tmodel = tmodel;
        this.row = row;
        int count = tmodel.getColumnCount() - 1;
        subItems.clear();
        for( int i = 0; i < count; i++ ){
            subItems.add( tmodel.getValueAt( row, i + 1 ) + "" );
        }
    }

    public String getName(){
        return name;
    }
    
    public void setName( String value ){
        name = value;
    }
    
    public Object clone(){
        updateStatusFromTableModel();
        int count = subItems.size();
        String[] values = new String[count];
        for( int i = 0; i < count; i++ ){
            values[i] = subItems.get( i );
        }
        BListViewItem ret = new BListViewItem( values );
        return ret;
    }
    
    public Object getTag(){
        return tag;
    }
    
    public void setTag( Object value ){
        tag = value;
    }
    
    public int getSubItemCount(){
        updateStatusFromTableModel();
        return subItems.size();
    }

    /**
     * テーブルモデルへの参照がある場合、テーブルモデルからデータを読み込み更新する
     */
    private void updateStatusFromTableModel(){
        if( tmodel == null ){
            return;
        }
        int count = tmodel.getColumnCount() - 1;
        if( subItems.size() != count ){
            subItems.clear();
            subItems.setSize( count );
        }
        for( int i = 0; i < count; i++ ){
            subItems.set( i, tmodel.getValueAt( row, i + 1 ) + "" );
        }
    }
    
    public String getSubItemAt( int index ){
        updateStatusFromTableModel();
        return subItems.get( index );
    }

    public void setSubItemAt( int index, String value ){
        updateStatusFromTableModel();
        subItems.set( index, value );
    }

}

File: org.kbinani.windows.forms/BListViewItem.cs
/*
 * BFontChooser.cs
 * Copyright (C) 2009-2010 kbinani
 *
 * This file is part of org.kbinani.windows.forms.
 *
 * org.kbinani.windows.forms is free software; you can redistribute it and/or
 * modify it under the terms of the BSD License.
 *
 * org.kbinani.windows.forms is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 */
#if JAVA
//INCLUDE ../BuildJavaUI/src/org/kbinani/windows/forms/BListViewItem.java
#else
namespace org.kbinani.windows.forms {

    public class BListViewItem : System.Windows.Forms.ListViewItem {
        private string group = "";

        public BListViewItem( string[] values )
            : base( values ) {
        }

        public BListViewItem()
            : base() {
        }

        public object clone() {
            return base.Clone();
        }

        public object getTag() {
            return base.Tag;
        }

        public void setTag( object value ) {
            base.Tag = value;
        }

        public int getSubItemCount() {
            return base.SubItems.Count;
        }

        public string getSubItemAt( int index ) {
            return base.SubItems[index].Text;
        }

        public void setSubItemAt( int index, string value ) {
            base.SubItems[index].Text = value;
        }

        public string getName() {
            return base.Name;
        }

        public void setName( string value ) {
            base.Name = value;
        }
    }

}
#endif
